src/model-helpers/Profile.js

```diff
@@ -0,0 +1,141 @@
+import {
+  buildThing,
+  deleteFile,
+  getStringNoLocale,
+  getThing,
+  getUrl,
+  getWebIdDataset,
+  removeStringNoLocale,
+  removeUrl,
+  saveFileInContainer,
+  saveSolidDatasetAt,
+  setThing
+} from '@inrupt/solid-client';
+import { RDF_PREDICATES } from '../constants';
+
+/**
+ * @typedef {import('@inrupt/solid-ui-react').SessionContext} Session
+ */
+
+/**
+ * A function fetches the user's profile information from their webId's profile
+ * card
+ *
+ * @function fetchProfileInfo
+ * @param {Session} session - Solid's Session Object {@link Session}
+ * @returns {Promise<object>} Object - The object containing the information related
+ * to the person on their profile card, the profile dataset, and the profile Thing
+ */
+export const fetchProfileInfo = async (session) => {
+  const profileDataset = await getWebIdDataset(session.info.webId);
+  const profileThing = getThing(profileDataset, session.info.webId);
+
+  const profileName = getStringNoLocale(profileThing, RDF_PREDICATES.profileName);
+  const nickname = getStringNoLocale(profileThing, RDF_PREDICATES.nickname);
+  const profileImage = getUrl(profileThing, RDF_PREDICATES.profileImg);
+
+  const profileInfo = { profileName, nickname, profileImage };
+
+  // TODO: include more fields to the object like organization, address, etc.
+  // when expanding this feature
+  return { profileInfo, profileDataset, profileThing };
+};
+
+/**
+ * A function that updates the user's profile information based on what is inputed
+ * in the Profile page
+ *
+ * @function updateProfileInfo
+ * @param {Session} session - Solid's Session Object {@link Session}
+ * @param {object} profileData - The object containing the information related
+ * to the person on their profile card, the profile dataset, and the profile Thing
+ * @param {object} inputValues - The inputs for updating profile information
+ * on their profile card
+ * @returns {Promise} Promise - Performs action to update profile card on the
+ * user's profile card
+ */
+export const updateProfileInfo = async (session, profileData, inputValues) => {
+  let { profileDataset, profileThing } = profileData;
+  const { profileInfo } = profileData;
+
+  Object.keys(inputValues).forEach((input) => {
+    switch (inputValues[input]) {
+      case null:
+      case 'null':
+      case 'No value set':
```

Comment:
What are these cases? It seems to me that if a person sets their name as `'no value set'` or `'null'`, there's no reason to say no.

For the case of `null`, it seems we can handle that the same as `''`

Reply:
Will remove cases for `"null"` and `"no value set"`. Because" no value set" is now just a placeholder, should be fine now compared to before.

For `null`, think I'm going to keep it since `''` doesn't fully cover the cases for when the property is null.

https://github.com/codeforpdx/PASS/assets/14917816/5fc843d3-3328-458c-b292-26ce4cffb589

If I were to were to remove it, it'll return the string `"null"` instead.

https://github.com/codeforpdx/PASS/assets/14917816/1a23a13b-de56-4737-a9fd-8d9002435361